homeworks/rostyslav.havuka_RostyslavHavuka/6-typeScript/src/index.ts

```diff
@@ -0,0 +1,247 @@
+const scoreElement = document.querySelector('[data-score]') as HTMLHeadingElement;
+const cubeElement = document.querySelector('[data-cube]') as HTMLDivElement;
+const cubeScoreElement = document.querySelector('[data-cube-score]') as HTMLDivElement;
+const keyElement = document.querySelector('[data-key]') as HTMLInputElement;
+const progressBarElement = document.querySelector('[data-progress-bar]') as HTMLDivElement;
+const keyStart = document.querySelector('[data-key-start]') as HTMLButtonElement
+const keyStop = document.querySelector('[data-key-stop]') as HTMLButtonElement
+const keyAgain = document.querySelector('[data-key-again]') as HTMLButtonElement
+
+class Game {
+  private currentKey: string = '';
+  private score: number = 100;
+  private interval: number = 2000;
+  private intervalId: NodeJS.Timeout;
+  private intervalStatusBarId: NodeJS.Timeout;
+  private gameState: boolean = false;
+  private timerProgressBar: number = 0;
+  private width: number;
+
+  constructor(
+    private scoreElement: HTMLHeadingElement,
+    private cubeScoreElement: HTMLDivElement,
+    private keyElement: HTMLInputElement,
+    private cubeElement?: HTMLDivElement,
+    private progressBarElement?: HTMLDivElement,
+  ) { }
+
+  start() {
+    window.addEventListener('load', this.startKeysInterval())
+
+    this.subscribeOnKeyPress()
+
+    keyStart.addEventListener('click', (): any => {
```

Comment:
Several points again:

1. You should pass `keyStart` as class param. Why? Your game class should be independent of DOM elements. When you use `keyStart` from a global variable, your game is linked to this element and you cannot init a new game with different elements or layouts. So please, add all keys elements as class params;

2. Create a separate method for each listener: `subsctibeOnKeyStartClick`, `subscribeOnKeyStopClick` .... etc.

3. **Don't use `any` in your code!** That's really bad practice and your code behaviors like a native JS code. So all benefits from TS are gone.